### DISCUSSION TITLE
Reduction of assumptions in (get-unsat-assumptions)

### DISCUSSION THREAD

**Discussion by grhcj** [cvc5/cvc5#10585] [Q&A] [answered]
We noticed that CVC5 (v1.1.0) returns the full set of assumptions when using (get-unsat-assumptions).
Consider the following SMT-Lib benchmark:

```
(set-logic QF_BVFP)
(set-option :produce-unsat-assumptions true)

(declare-fun a () Bool)
(declare-fun b () Bool)
(declare-fun c () Bool)
(declare-fun d () Bool)
(declare-fun e () Bool)

(assert (or a b))
(assert (or a (not b)))
(assert (or (or c d) e))

(check-sat-assuming ((not a) c d e))
(get-unsat-assumptions)
```

Bitwuzla, Mathsat 5 and Z3 all return ((not a)) as unsatisfiable assumption set.
CVC5 returns ((not a) c d e).
Is this expected behaviour in CVC5? Is it possible to retrieve a reduced assumption set?

Thanks in advance
Lukas

**Comment by daniel-larraz** [comment, marked as answer]
Answered in issue #10619
